WordPressAuthenticator/Navigation/NavigateToEnterSiteCredentials.swift

```diff
@@ -0,0 +1,28 @@
+import Foundation
+
+/// Navigates to the wp-admin site credentials flow.
+///
+public struct NavigateToEnterSiteCredentials: NavigationCommand {
+    private let loginFields: LoginFields
+
+    public init(loginFields: LoginFields) {
+        self.loginFields = loginFields
+    }
+    public func execute(from: UIViewController?) {
+        let navigationController = (from as? UINavigationController) ?? from?.navigationController
+        presentSiteCredentialsView(navigationController: navigationController,
+                                   loginFields: loginFields)
+    }
+}
+
+private extension NavigateToEnterSiteCredentials {
+    func presentSiteCredentialsView(navigationController: UINavigationController?, loginFields: LoginFields) {
+        guard let controller = SiteCredentialsViewController.instantiate(from: .siteAddress) else {
+            WPAuthenticatorLogError("Failed to navigate to SiteCredentialsViewController")
+            return
+        }
+
+        controller.loginFields = loginFields
```

Comment:
Jaclyn left some comments regarding the need to initialize or not both the `dismissBlock` and `errorToPresent`, I'm unsure what possible scenarios could require those two values since we're jumping directly into them from the deep link data and how we could feed those two fields from WciOS. 

With that said, I just went without initializing those two fields. Please let me know if you think we should do otherwise.

Reply:
From what I understand, `errorToPresent` is internal to the authenticator library, it is used to handle login errors only not any other errors from outside.

`dismissBlock` is only called when the `SiteCredentialsViewController` is initialized with `isDismissible: true`, which will show a dismiss button instead of the back button on the top left. This is not relevant for our case as we are pushing the view into the navigation stack now.

Reply:
Thank you for this clarification, @itsmeichigo! 🙇  I came to a similar conclusion but got worried I was missing something I didn't understand completely. 😅